server/src/test/java/org/elasticsearch/search/aggregations/bucket/terms/StringTermsTests.java

```diff
@@ -34,13 +34,26 @@
 import java.util.Set;
 
 public class StringTermsTests extends InternalTermsTestCase {
+    private BytesRef[] dict;
+
+    public synchronized void generateRandomDict() {
+        if (dict == null) {
+            Set<BytesRef> terms = new HashSet<>();
+            int numTerms = randomIntBetween(2, 100);
+            for (int i = 0; i < numTerms; i++) {
+                terms.add(new BytesRef(randomAlphaOfLength(10)));
+            }
+            dict = terms.stream().toArray(BytesRef[]::new);
+        }
+    }
 
     @Override
     protected InternalTerms<?, ?> createTestInstance(String name,
                                                      Map<String, Object> metadata,
                                                      InternalAggregations aggregations,
                                                      boolean showTermDocCountError,
                                                      long docCountError) {
+        generateRandomDict();
```

Comment:
Can `dict` be a local variable?

Reply:
I want to ensure that there are some equal keys in the instances that we try to merge so I build a single dictionary for the entire test suite. It could be easier if the class was not used by the AggregationTests.

Reply:
Could you override `randomResultsToReduce`?

Reply:
++,  I pushed https://github.com/elastic/elasticsearch/pull/64469/commits/3777760966b73941679ef742ce86e7149c3bf6dd